src/ejabberd_gen_sm.erl

```diff
@@ -80,9 +80,9 @@ update_session(Mod, User, Server, Resource, Session) ->
 delete_session(Mod, Sid, User, Server, Resource) ->
     Mod:delete_session(Sid, User, Server, Resource).
 
--spec cleanup(Mod :: module(), Node :: atom()) -> any().
-cleanup(Mod, Node) ->
-    Mod:cleanup(Node).
+-spec cleanup(Mod :: module(), Node :: atom(), Initial :: boolean()) -> any().
+cleanup(Mod, Node, Initial) ->
+    Mod:cleanup(Node, Initial).
```

Comment:
I think it would be better to leave this interface as it was before. The motivation is that Redis backend is the only one doing the initial cleanup, so I would only have this two-argument function as a local function in `ejabberd_sm_redis`.